launch/launch/substitutions/string_join_substitution.py

```diff
@@ -76,6 +78,18 @@ def delimiter(self) -> List[Substitution]:
         """Getter for delimiter."""
         return self.__delimiter
 
+    @classmethod
+    def parse(cls, data: Sequence[SomeSubstitutionsType]):
+        """Parse `StringJoinSubstitution` substitution."""
+        if len(data) < 2:
+            raise TypeError(
+                'string-join substitution expects at least 2 arguments'
```

Comment:
It might be worth expanding a bit, e.g.

```suggestion
                'string-join substitution expects at least 2 arguments: '
                '1 delimiter + at least 1 component'
```